quinn-proto/src/endpoint.rs

```diff
@@ -1052,6 +1050,23 @@ pub enum ConnectError {
     UnsupportedVersion,
 }
 
+/// Error for attempting to retry an [`IncomingConnection`] that can not be retried
+#[derive(Debug, Error)]
+pub struct RetryError(pub IncomingConnection);
```

Comment:
Does `pub` here add much value, given that `into_incoming` exists? If not, let's keep the field private to reserve more space for non-breaking changes in the future.

Reply:
Made private. I sometimes like pattern-destructuring things like this, but it's fine.